src/mocompiler.js

```diff
@@ -258,8 +267,8 @@ Compiler.prototype._build = function (list, size) {
   for (i = 0, len = list.length; i < len; i++) {
     const msgidLength = /** @type {Buffer} */(/** @type {unknown} */(list[i].msgid));
     msgidLength.copy(returnBuffer, curPosition);
-    returnBuffer.writeUInt32LE(list[i].msgid.length, 28 + i * 8);
```

Comment:
ah thank you for fixing this. i had forgotten to revert the changes for those lines